### ISSUE TITLE
Header inclusion pulls in a lot of the C library, unnecessarily.

### ISSUE THREAD

**Issue by jmorse** [esbmc/esbmc#84]
That is; if you include stdio.h say, then there are references everywhere for all the IO functions, and so they get pulled into the final program.

Possibly instead the set of functions that are called in should be used to work out how much of the C library to include.

**Comment by rafaelsamenezes** [comment]
@mikhailramalho Is this still relevant with the clang frontend?

**Comment by mikhailramalho** [comment]
Yes, you can see that we pull a bunch of unused stuff from the model when verifying programs.

The code in `generate_symbol_deps` needs to be reviewed.
